### ISSUE TITLE
Do not initialize file logger before reading custom config

### ISSUE THREAD

**Issue by discordianfish** [grafana/grafana#3642]
Hi,

apparently grafana initializes the default logger before reading the custom config (`-config`).
If you for example disable the file logger or change it's path in the custom config, grafana still starts with the defaults and attempts to write to data/log which, if it fails, prevents startup.

Instead grafana should first read the config before using any configurable options. Alternatively to that, it would be already an improvement to use more sane default like using only the console logger by default.

**Comment by torkelo** [comment]
you can change log file path using the command line, cfg:default.paths.logs=/var/log/grafana/log

The reason for initialising loging before reading config file is in order to log error when reading config file.

**Comment by discordianfish** [comment]
@torkelo I figured that out by now, but consider this being a pretty bad user experience: I spent quite some time figuring out why it doesn't seem to read my config file until I read the code and figured out it's using the default values at start.
I know it's a chicken-egg problem regarding logging errors reading config files, but I'd argue that it's less surprising if it would simply log to console which should always work. But sure, that's your decision. I got my deployment working :).
